### PR TITLE
Add BuildUniversalReferenceDocument step and some refactor

### PR DESCRIPTION
**Pull request by hellosnow**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.ManagedReference/BuildManagedReferenceDocument.cs [hunk 1]
```diff
@@ -22,38 +17,25 @@ public class BuildManagedReferenceDocument : BuildReferenceDocumentBase, ISuppor
             new CompositeModelAttributeHandler(
                 new UniqueIdentityReferenceHandler(),
                 new MarkdownContentHandler()
-                );
+            );
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildManagedReferenceDocument.cs]: move to base?

Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildManagedReferenceDocument.cs]: what if child has different handlers? Or make the handlers and context as default?

---
In reply to: [114489466](https://github.com/dotnet/docfx/pull/1621#discussion_r114489466) [](ancestors = 114489466)

File: plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildUniversalReferenceDocument.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            new CompositeModelAttributeHandler(
+                new UniqueIdentityReferenceHandler(),
+                new MarkdownContentHandler()
+            );
+
+        public override string Name => nameof(BuildUniversalReferenceDocument);
+
+        #region BuildReferenceDocumentBase
+
+        protected override void BuildArticle(IHostService host, FileModel model)
```
Comment (vicancy) [inline plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildUniversalReferenceDocument.cs]: >BuildArticle [](start = 32, length = 12)

can move to base?

Comment (hellosnow) [inline plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildUniversalReferenceDocument.cs]: this is an abstract method to be implemented in child?

---
In reply to: [114690938](https://github.com/dotnet/docfx/pull/1621#discussion_r114690938) [](ancestors = 114690938)

Comment (vicancy) [inline plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildUniversalReferenceDocument.cs]: change to virtual?

---
In reply to: [114691552](https://github.com/dotnet/docfx/pull/1621#discussion_r114691552) [](ancestors = 114691552,114690938)

File: plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildUniversalReferenceDocument.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        public override string Name => nameof(BuildUniversalReferenceDocument);
+
+        #region BuildReferenceDocumentBase
+
+        protected override void BuildArticle(IHostService host, FileModel model)
+        {
+            var pageViewModel = (PageViewModel)model.Content;
+
+            HandleAttributes(host, model, shouldSkipMarkup: pageViewModel?.ShouldSkipMarkup ?? false);
```
Comment (vicancy) [inline plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildUniversalReferenceDocument.cs]: >HandleAttributes [](start = 12, length = 16)

naming: BuildArticleCore?

**Comment by vicancy** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by vwxyzh** [comment]
:shipit:

**Pull request merged**